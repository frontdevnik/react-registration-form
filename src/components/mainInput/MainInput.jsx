import React from 'react';
import classnames from 'classnames';

export default ({
  type,
  name,
  required,
  className,
  labelText,
  placeholder,
  pattern,
  errorMessage,
  children,
  register,
  error,
}) => {
  return (
    <div className={classnames('input-field col s12', className)}>
      <input
        name={name}
        type={type}
        id={name}
        placeholder={placeholder}
        className={error ? 'invalid' : ''}
        ref={register({
          required: required && 'This is required field',
          pattern: {
            value: pattern,
            message: errorMessage,
          },
        })}
      />
      <label htmlFor={name}>{labelText}</label>
      {error && (
        <span className="helper-text" data-error={error}>
          {error}
        </span>
      )}
      {children}
    </div>
  );
};
