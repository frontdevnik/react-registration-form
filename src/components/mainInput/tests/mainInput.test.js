import React from 'react';

import MainInput from '../MainInput';

import { stubInputProps } from './stubProps';

const setUp = (props) => shallow(<MainInput {...props} />);

describe('<MainInput />', () => {
  let component;

  beforeEach(() => {
    component = setUp(stubInputProps);
  });

  it('should render correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('should main div has passed class', () => {
    expect(
      component.find('div').hasClass(stubInputProps.className)
    ).toBeTruthy();
  });

  describe('<input /> has correct props', () => {
    let component;
    let input;

    beforeEach(() => {
      component = setUp(stubInputProps);
      input = component.find('input');
    });

    it('should input has correct name', () => {
      expect(input.props().name).toBe(stubInputProps.name);
    });

    it('should input has correct type', () => {
      expect(input.props().type).toBe(stubInputProps.type);
    });

    it('should input has correct id', () => {
      expect(input.props().id).toBe(stubInputProps.name);
    });

    it('should input has valid class', () => {
      component = setUp({ ...stubInputProps, error: '' });
      input = component.find('input');

      expect(input.hasClass('')).toBeTruthy();
    });

    it('should input has invalid class', () => {
      expect(input.hasClass('invalid')).toBeTruthy();
    });
  });

  it('should label has correct htmlFor props', () => {
    expect(component.find('label').props().htmlFor).toBe(stubInputProps.name);
  });

  it('should label has correct text', () => {
    expect(component.find('label').text()).toBe(stubInputProps.labelText);
  });

  it('should render helper-text with error state', () => {
    expect(component.find('span').text()).toBe(stubInputProps.error);
  });
});
