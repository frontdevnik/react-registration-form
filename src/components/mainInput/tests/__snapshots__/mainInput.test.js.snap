// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainInput /> should render correctly 1`] = `"<div class=\\"input-field col s12 active\\"><input type=\\"text\\" name=\\"test\\" id=\\"test\\" placeholder=\\"placeholder\\" class=\\"invalid\\"/><label for=\\"test\\">label</label><span class=\\"helper-text\\" data-error=\\"error\\">error</span></div>"`;
